### PR TITLE
Feature/20995 delete order data capture

### PR DESCRIPTION
**Pull request by dssherlock**

### CODE REVIEW

File: src/NHSD.GPIT.BuyingCatalogue.Services/Orders/OrderAdminService.cs [hunk 1]
```diff
@@ -134,15 +135,28 @@ public async Task<IList<SearchFilterModel>> GetOrdersBySearchTerm(string search)
                 .ToList();
         }
 
-        public async Task DeleteOrder(CallOffId callOffId)
+        public async Task DeleteOrder(CallOffId callOffId, string nameOfRequester, string nameOfApprover, DateTime? dateOfApproval)
         {
+            callOffId.ValidateNotNull(nameof(callOffId));
```
Comment (ghost) [inline src/NHSD.GPIT.BuyingCatalogue.Services/Orders/OrderAdminService.cs]: `CallOffId` is a struct so it should never be null unless explicitly used as a nullable type `Nullable<CallOffId>` or `CallOffId?`

**Review by bjss-colin — COMMENTED** [review]
All looks good, just one comment on the `OrderAdminService`

File: src/NHSD.GPIT.BuyingCatalogue.Services/Orders/OrderAdminService.cs (hunk 1 above)
Comment (dssherlock) [reply src/NHSD.GPIT.BuyingCatalogue.Services/Orders/OrderAdminService.cs]: Yep good spot-i must admit i hadnt looked into this too much I just threw it in when i added the new ones. But yes u r right

**Review by imranbashir01 — DISMISSED** [review]

File: src/NHSD.GPIT.BuyingCatalogue.EntityFramework/Ordering/Configuration/OrderDeletionApprovalEntityTypeConfiguration.cs [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    {
+        public void Configure(EntityTypeBuilder<OrderDeletionApproval> builder)
+        {
+            builder.ToTable("OrderDeletionApprovals", Schemas.Ordering);
+
+            builder.HasKey(a => a.Id).HasName("PK_OrderDeletionApproval");
+
+            builder.Property(x => x.OrderId).IsRequired();
+
+            builder.Property(a => a.NameOfApprover);
```
Comment (magnusjohn-bjss) [inline src/NHSD.GPIT.BuyingCatalogue.EntityFramework/Ordering/Configuration/OrderDeletionApprovalEntityTypeConfiguration.cs:19]: These three properties are marked not null in the db, so should have .IsRequired() added to their config

**Review by imranbashir01 — DISMISSED** [review]

**Review by bjss-colin — DISMISSED** [review]

**Review by bjss-colin — APPROVED** [review]

**Review by imranbashir01 — APPROVED** [review]

**Pull request merged**